lib/concurrent/synchronization/abstract_struct.rb

```diff
@@ -6,9 +6,18 @@ module Synchronization
     module AbstractStruct
 
       # @!visibility private
-      def initialize(*values)
+      def initialize(*values, **kw_values)
         super()
-        ns_initialize(*values)
+        ns_initialize(*values, **kw_values)
+      end
+
+      # @!macro [attach] struct_keyword_init
```

Comment:
The macro definition at the beginning of the can doc can be omitted, if the doc is not shared with other methods.